src/embedding/kd.ts

```diff
@@ -0,0 +1,414 @@
+// (C) 2020 GoodData Corporation
+import {
+    IGdcMessageEvent,
+    getEventType,
+    GdcProductName,
+    IGdcMessageEnvelope,
+    IDrillableItemsCommandBody
+} from './common';
+
+export namespace EmbeddedKpiDashboard {
+    /**
+     * Base type for KD events.
+     */
+    export type IKdMessageEvent<T, TBody> = IGdcMessageEvent<GdcProductName.KPI_DASHBOARD, T, TBody>;
+
+    /**
+     * Base type for KD event data.
+     */
+    export type IKdGdcMessageEnvelope<T, TBody> = IGdcMessageEnvelope<GdcProductName.KPI_DASHBOARD, T, TBody>;
+
+    /**
+     * All KD command Types.
+     */
+    export enum GdcKdCommandType {
+        /**
+         * The command save a dashboard.
+         */
+        Save = 'saveDashboard',
+
+        /**
+         * The command cancel editing dashboard.
+         */
+        CancelEdit = 'cancelEdit',
+
+        /**
+         * The command delete existed dashboard.
+         */
+        Delete = 'deleteDashboard',
+
+        /**
+         * The command edit a dashboard.
+         */
+        SwitchToEdit = 'switchToEdit',
+
+        /**
+         * The command set drillable items.
+         */
+        DrillableItems = 'drillableItems',
+
+        /**
+         * The command set size of dashboard.
+         */
+        SetSize = 'setSize'
+    }
+
+    /**
+     * All KD event types.
+     */
+    export enum GdcKdEventType {
+        /**
+         * Type represent that the dashboard listening for drilling event.
+         */
+        ListeningForDrillableItems = 'listeningForDrillableItems',
+
+        /**
+         * Type represent that the embedded content starts loading.
+         */
+        LoadingStarted = 'loadingStarted',
+
+        /**
+         * Type represent that The user does not have permissions to view or edit the content.
+         */
+        NoPermissions = 'noPermissions',
+
+        /**
+         * Type represent that an operation increasing the height of the hosting iframe is performed.
+         */
+        Resized = 'resized',
```

Comment:
missing typing for `ResizedData`. should it include what sized is applied?

Reply:
updated